src/interpreter/example-validator.ts

```diff
@@ -229,6 +232,10 @@ export class ExampleValidator implements ProfileAstVisitor {
     return isValid;
   }
 
+  visitComlinkNoneLiteralNode(_node: ComlinkNoneLiteralNode): boolean {
+    return true;
```

Comment:
question: shouldn't this do something? I _feel_ like it should validate that the field it is being assigned to is either not required or optional, similar to `visitComlinkPrimitiveLiteralNode`. 

Looking at the spaghetti code though, I think it might be hard due to how the state is currently propagated around the validator and especially how [NonNull](https://github.com/superfaceai/parser/blob/4643acc167be193e1ea7374660817be4e15f95ed/src/interpreter/example-validator.ts#L444) is somehow unwraped here for some reason.

Reply:
I think you were right. It can at least check that we are not trying to use None for `NonNullStructure`.